client/windows/CCastleInterface.h

```diff
@@ -393,8 +394,11 @@ class CMageGuildScreen : public CStatusbarWindow
 
 	std::shared_ptr<CMinorResDataBar> resdatabar;
 
+	ObjectInstanceID townId;
+
 public:
 	CMageGuildScreen(CCastleInterface * owner, const ImagePath & image);
+	void update();
```

Comment:
1) More clear name as to what changed
2) Method needs to check whether it is our town changed or some other one, e.g. another player in simturns
```suggestion
	void townSpellsChanged(ObjectInstanceID townID);
```

Reply:
Should be irrelevant. This is for opened window. And all opened mage guild windows will be updated, when there was an spell change.

Reply:
No. 
If player A updates spell in his town X, player B does not needs an update if he is looking at his mages guild in town Y.

Reply:
Changed